results-processor/main.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python3
+import re
+import shutil
+import tempfile
+
+import flask
+from google.cloud import storage
+
+import wptreport
+
+
+APPENGINE_INTERNAL_IP = '10.0.0.1'
+
+
+app = flask.Flask(__name__)
+
+
+@app.route('/api/results/process', methods=['POST'])
+def task_handler():
+    if flask.request.remote_addr != APPENGINE_INTERNAL_IP:
+        return 'External requests not allowed', 403
+
+    params = flask.request.form
+    # Mandatory fields:
+    # uploader = params['uploader']
+    gcs_path = params['gcs']
+    result_type = params['type']
+    # TODO(Hexcles): Support multiple results.
+    assert result_type == 'single'
+    # Optional fields (to be deprecated once everyone has embedded metadata):
+    browser = params.get('browser')
+    revision = params.get('revision')
+
+    match = re.match(r'/([^/]+)/(.*)', gcs_path)
+    assert match
+    bucket_name, blob_path = match.groups()
+
+    gcs = storage.Client()
+    bucket = gcs.get_bucket(bucket_name)
+    blob = bucket.blob(blob_path)
+
+    with tempfile.NamedTemporaryFile(suffix='.json') as temp:
+        blob.download_to_file(temp)
+        temp.seek(0)
+        report = wptreport.WPTReport()
+        report.load_json(temp)
+
+    resp = "{} results loaded from {}".format(len(report.results), gcs_path)
```

Comment:
Is `"{} and {}".format(foo, bar)` somehow preferable to `"%s and %s" % (foo, bar)` ?

Reply:
Not really in this case, but `"{}".format` has some more advanced features (like named arguments) which aren't used here.